webanno-api-dao/src/main/java/de/tudarmstadt/ukp/clarin/webanno/api/dao/CasStorageServiceImpl.java

```diff
@@ -999,9 +1007,17 @@ public void close()
         {
             if (holder != null) {
                 try {
-                    log.trace("Returning briefly borrowed CAS [{}]@[{}]({})", username,
-                            documentName, documentId);
-                    exclusiveAccessPool.returnObject(key, holder);
+                    if (invalidateOnClose) {
+                        log.trace("Invalidating CAS [{}]@[{}]({})", username,
+                                documentName, documentId);
+                        exclusiveAccessPool.invalidateObject(key, holder);
```

Comment:
Why do we need to call `exclusiveAccessPool.invalidateObject(key, holder);` here if it is already called here?

https://github.com/webanno/webanno/pull/1750/files#diff-71ecaac6a6015990de0c535cc226f8b4R743-R744

Reply:
I'm setting a flag to call invalidate when closing the exclusiveaccess. How I understand it: with `clear` in the delete method we will only delete idle objects and the cas we are trying to delete is still allocated because we borrowed it with the exclusiveaccess. But if we invalidate it at the same time in the delete method, we will have problems when trying to put back on access.close. So, it made the most sense to me to tell the access with the flag to invalidate the object on close.